src/config.rs

```diff
@@ -144,48 +146,49 @@ impl Dependency {
+                );
+                table.insert(
+                    "git",
+                    value(&dep.git).into_value().expect("git URL should be a valid toml value"),
+                );
+
+                match &dep.tag {
+                    Some(tag) => {
                         table.insert(
-                            "version",
-                            value(&dep.version)
-                                .into_value()
-                                .expect("version should be a valid toml value"),
+                            "tag",
+                            value(tag).into_value().expect("tag should be a valid toml value"),
                         );
-                        table.insert(
-                            "git",
-                            value(&dep.git)
-                                .into_value()
-                                .expect("git URL should be a valid toml value"),
-                        );
-                        table.insert(
-                            "rev",
-                            value(rev).into_value().expect("rev should be a valid toml value"),
-                        );
-                        value(table)
                     }
-                    None => {
-                        let mut table = InlineTable::new();
-                        table.insert(
-                            "version",
-                            value(&dep.version)
-                                .into_value()
-                                .expect("version should be a valid toml value"),
-                        );
-                        table.insert(
-                            "git",
-                            value(&dep.git)
-                                .into_value()
-                                .expect("git URL should be a valid toml value"),
-                        );
-
-                        value(table)
-                    }
-                },
-            ),
+                    None => match &dep.branch {
+                        Some(branch) => {
+                            table.insert(
+                                "branch",
+                                value(branch)
+                                    .into_value()
+                                    .expect("branch should be a valid toml value"),
+                            );
+                        }
+                        None => match &dep.rev {
+                            Some(rev) => {
+                                table.insert(
+                                    "rev",
+                                    value(rev)
+                                        .into_value()
+                                        .expect("rev should be a valid toml value"),
+                                );
+                            }
+                            None => {}
+                        },
+                    },
+                };
```

Comment:
Avoid nested match statements. Much better to match on a tuple: `match (&dep.tag, &dep.branch, &dep.rev) { (Some(tag), None, None) => { /* ... */ } }`